src/libraries/System.Text.Json/tests/System.Text.Json.FSharp.Tests/EnumTests.fs

```diff
@@ -58,3 +72,18 @@ let ``Fail Serialize Good Value Of Bad Enum Type`` () =
     let ex = Assert.Throws<TargetInvocationException>(fun () ->  JsonSerializer.Serialize(badEnumWithGoodValue, options) |> ignore)
     Assert.Equal(typeof<InvalidOperationException>, ex.InnerException.GetType())
     Assert.Equal("Enum type 'BadEnum' uses unsupported identifer name 'There's a comma, in my name'.", ex.InnerException.Message)
+
+[<Fact>]
+let ``Fail Deserialize Numeric label Of Enum When Disallow Integer Values`` () =
+    Assert.Throws<JsonException>(fun () -> JsonSerializer.Deserialize<NumericLabelEnum>(enumWithNumericValueJsonStr, optionsDisableNumeric) |> ignore)
```

Comment:
Initially, I would have expected we'd try to make this succeed if the numeric value matches the label but probably it's too niche for us to care :-)